src/common/test/MockApolloClientProvider.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { ApolloProvider } from '@apollo/client';
+import { GlobalWithFetchMock } from 'jest-fetch-mock/types';
+import { GraphQLError } from 'graphql';
+
+import graphqlClient from '../../graphql/client';
+import { ProfileData, UpdateProfileData } from '../../graphql/typings';
+
+export type MockedResponse = {
+  profileData?: ProfileData;
+  updatedProfileData?: UpdateProfileData;
+  errorType?: 'networkError' | 'graphQLError';
+};
+
+type RequestVariables = Record<string, unknown>;
+export type ResponseProvider = (variables?: RequestVariables) => MockedResponse;
+
+export function MockApolloClientProvider({
+  responseProvider,
+  children,
+}: {
+  children: React.ReactElement | React.ReactNodeArray;
+  responseProvider: ResponseProvider;
+}): React.ReactElement {
+  const customGlobal = global as GlobalWithFetchMock;
+  customGlobal.fetchMock.mockResponse((variables?: RequestVariables) => {
+    const response = createMockedProfileResponse(responseProvider(variables));
+    if (response.error) {
+      return Promise.reject({
+        body: JSON.stringify({ message: (response.error as Error).message }),
+      });
+    }
+    return Promise.resolve({ body: JSON.stringify(response) });
+  });
+  return <ApolloProvider client={graphqlClient}>{children}</ApolloProvider>;
+}
+
+const createMockedProfileResponse = (
+  response: MockedResponse
+): Record<string, unknown> => {
+  if (!response) {
+    throw new Error('No response provided');
+  }
+  const { errorType, profileData, updatedProfileData } = response;
+  const data = !errorType
+    ? profileData
+      ? { myProfile: profileData }
+      : { updateMyProfile: { profile: updatedProfileData } }
+    : undefined;
+  return !errorType
+    ? {
+        data,
+      }
+    : errorType === 'networkError'
+    ? { error: new Error(`NetworkError at ${Date.now()}`) }
+    : {
+        errors: [new GraphQLError(`GraphQLError at ${Date.now()}`)],
+      };
```

Comment:
Quite a many ternary operators here. I find it somewhat hard to read. Changing some of them to `if..elses` could IMO improve readability.

```suggestion
  if (!errorType) {
    const data = profileData
      ? { myProfile: profileData }
      : { updateMyProfile: { profile: updatedProfileData } };
    return { data };
  } else if (errorType === 'networkError') {
    return { error: new Error(`NetworkError at ${Date.now()}`) };
  } else {
    return {
      errors: [new GraphQLError(`GraphQLError at ${Date.now()}`)],
    };
  }
```

Reply:
Changed ternary jungle into separate functions